apps/src/templates/manageStudents/manageStudentsRedux.js

```diff
@@ -376,6 +378,17 @@ export default function manageStudents(state=initialState, action) {
     }
     return newState;
   }
+  if (action.type === UPDATE_ALL_SHARE_SETTING) {
+    let newState = {
```

Comment:
If not all the students are already in edit mode, I don't think this will work. There won't be any `editingData` for every student id to update, which will cause an error on line 388 of `manageStudentsRedux`. I think you need to dispatch `EDIT_ALL` before `UPDATE_ALL_SHARE_SETTING ` in the function on line 101.

something like this?

```
export const updateAllShareSetting = (disable) => {
   return (dispatch, getState) => {
     dispatch(editAll());
     dispatch({type: UPDATE_ALL_SHARE_SETTING, disable});
   };
 };
```